packages/schematics/angular/app-shell/index.ts

```diff
@@ -282,12 +300,15 @@ function addShellComponent(options: AppShellOptions): Rule {
 }
 
 export default function (options: AppShellOptions): Rule {
-  return chain([
-    addUniversalApp(options),
-    addAppShellConfig(options),
-    addRouterModule(options),
-    addRouterOutlet(options),
-    addServerRoutes(options),
-    addShellComponent(options),
-  ]);
+  return (host: Tree, context: SchematicContext) => {
+    validateProject(host, context, options);
```

Comment:
This could be a rule, simplifying this whole thing:

```typescript
  return chain([
    (host: Tree, context: SchematicContext) => validateProject(host, context, options),
    ...
  ]);
```

Or even better, have `validateProject` be a `RuleFactory` which takes an options object and returns a Rule.